.github/workflows/build_pr_cache.yml

```diff
@@ -0,0 +1,66 @@
+# Simple workflow for building sccache for PR testing
+name: Build cache for PR testing
+# Note this will currently create a new sscache file and this must be manually pruned until such time as we have a job
+# for this.
+on:
+  # Uncomment to test this path on a pull request
+  # pull_request:
+  #   paths:
+  #     .github/workflows/build_pr_cache.yml
+  #     .github/do_build_ock/**  
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/build_pr_cache.yml
+      - .github/actions/do_build_ock/**
+      - .github/actions/setup_ubuntu_build/**
+  
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build_ca_host_x86_64:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      
+      # installs tools, ninja and installs llvm (default 16, RelAssert) and sets up cache
+      - name: setup-ubuntu
+        uses:  ./.github/actions/setup_ubuntu_build
+        with:
+          save: true
+
+      - name: build_host_x86_64-release
+        uses: ./.github/actions/do_build_ock
+        with:
+          build_type: Release
+
+      - name: build_host_x86_64-offline
+        uses: ./.github/actions/do_build_ock
+        with:
+          build_type: Release
+          extra_flags: -DCA_RUNTIME_COMPILER_ENABLED=OFF -DCA_EXTERNAL_CLC=${{ github.workspace }}/build/bin/clc
+          build_dir: build_offline
+          build_targets: UnitCL
+
+      - name: run host offline for extra checking
+        run:
+          ninja -C build check-UnitCL
+      
+      - name: clean_build
+        run:
+          rm -rf ${{ github.workspace }}/build*
+
+      - name: build riscv m1
+        uses: ./.github/actions/do_build_ock/do_build_m1  
+
+      - name: run riscv M1 UnitCL for extra checking
+        run:
+          ninja -C build check-UnitCL
```

Comment:
What does running `check-UnitCL` and `check-all-lit` actually do when building a cache? We've already built UnitCL previously, haven't we? Wouldn't this just pointlessly run tests?

Reply:
we are, it was more giving us confidence it was doing the right thing. I was in two minds as to the benefit, so I'll remove